crates/common/consensus/src/helpers.rs

```
@@ -98,3 +104,58 @@ pub fn get_voting_source(store: &Store, block_root: B256) -> Checkpoint {
         head_state.current_justified_checkpoint
     }
 }
+
+pub fn is_valid_indexed_attestation(
+    state: &BeaconState,
+    indexed_attestation: &IndexedAttestation,
+) -> anyhow::Result<bool> {
+    let indices: Vec<usize> = indexed_attestation
+        .attesting_indices
+        .iter()
+        .map(|&i| i as usize)
+        .collect();
+
+    if indices.is_empty() || !is_sorted_and_unique(&indices) {
+        return Ok(false);
+    }
+
+    let pubkeys: Vec<PubKey> = indices
+        .iter()
+        .filter_map(|&i| state.validators.get(i).map(|v| v.pubkey.clone()))
+        .collect();
+
+    let domain = match state.get_domain(
+        DOMAIN_BEACON_ATTESTER,
+        Some(indexed_attestation.data.target.epoch),
+    ) {
+        Ok(domain) => domain,
+        Err(_) => return Ok(false),
+    };
+
+    let signing_root = compute_signing_root(&indexed_attestation.data, domain);
+
+    let sig = blst::min_pk::Signature::from_bytes(&indexed_attestation.signature.signature)
+        .map_err(|err| anyhow!("Signarure conversion failed:{:?}", err))?;
+
+    let publickeys: Vec<blst::min_pk::PublicKey> = pubkeys
+        .iter()
+        .filter_map(|key| blst::min_pk::PublicKey::from_bytes(&key.inner).ok())
+        .collect();
```

If one of these public keys fails to convert we should return an error, not continue with the ones that did work.

---

checking